dali/kernels/imgproc/surface.h

```diff
@@ -28,6 +28,18 @@ struct Surface2D {
   __host__ __device__ constexpr T &operator()(int x, int y, int c = 0) const {
     return data[y * row_stride + x * pixel_stride + c * channel_stride];
   }
+
+  template <typename U = T,
```

Comment:
Please put a comment here, that those template incantations are just promotions to Surface to const.